README.md

```diff
@@ -4,12 +4,141 @@
+          - Kafka start up also has a deamon mode.
+    - Right now, there is no Kafka GUI (can still use the CLI tools, located in `/usr/bin` where you installed Kafka)
+    - To check if ZooKeeper is running and OK:
+        - ```$ echo 'ruok' | nc localhost 2181``` should return `imok`
+  - **Spark:**
+    - Manual installation:
+      - Config for .profile SPARK_HOME, JAVA_HOME, add all of these to $PATH
+      - Go to spark.apache.org
+      - Download the binary for "2.2.0, Pre-built for Apache Hadoop 2.7"
+    - Sbin within script: `sbin/start-master.sh` (dependent on where you installed Spark locally)
+      - Then you can see localhost:8080 for the Spark dashboard (to see Spark Master)
+      - There is also an `sbin/start-all.sh` to start the master and “agent” (slave) systems.
+    - About Spark cluster management / cluster environment:
+      - Master node controls the rest
+        - Master process
+        - Worker process
+        - On the worker process you launch executors, and parallelization occurs there
+      - Executor nodes (multiple) run the job(s)
+        - Where applications run
+        - Executors are executed by workers
+      - Driver is your application, where you write the code
+        - One that sends work to Executors
+        - Driver is on another executor node
+        - Your program; Driver is the program that can be run wherever
+        - Has implications on how data moves around depending for the operation you’re running
+      - Start slave doesn’t start unless you point to something
+          - Start-all doesn’t require that
+4. Launch the Application
+  - Deploy the App to Spark: `spark-submit --class EstimatorStreamingApp --name EstimatorStreamingApp --master spark:/SPARK-MASTER-URL.local:7077 --deploy-mode cluster --executor-memory 14G --num-executors 2`
+    - `Spark-job —class Class of App —name Name of App —master master node URL —deploy-mode cluster (still even running locally) — executor-memory 14G (if not available, will use less) —num-executors`
+    - you can retrieve the Spark Master and Slave Node URLs from the GUI
+    - Local / local-cluster / cluster options: deploy-mode has nothing to do with how you run the application, its only for launching the application.
+  - Start the Application via the generated Jar: `.../ld4p-data-pipeline/EstimatorStreamingApp/target/scala-2.11/EstimatorStreamingApp-assembly-1.0.0-SNAPSHOT.jar`
+    - change this to wherever you downloaded the codebase and built the Jar
+    - For Darren, had a permission issue on logging for the app, so used `sudo`, e.g.
+      - `sudo /opt/spark/bin/spark-submit --class EstimatorStreamingApp --name EstimatorStreamingApp --master spark://sul-dlweber-ubuntu:7077  --deploy-mode cluster --executor-memory 6G --num-executors 1`
+      - `/data/src/dlss/ld4l/ld4p-data-pipeline/EstimatorStreamingApp/target/scala-2.11/EstimatorStreamingApp-assembly-1.0.0-SNAPSHOT.jar`
+5. Create a Kafka Topic for the MARC
```

Comment:
Move 5. to 4. or just the part about creating the Kafka topic with the right number of partitions.  Without that, the default partitions will be 1 and that's not enough to parallelize the spark consumers/tasks.